src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -118,26 +126,29 @@ def delete_sub_collection_in_batches(
         sub_collection_ref (firestore.CollectionReference): The reference to the sub collection
         """
         try:
-            docs = sub_collection_ref.limit(self.DELETE_BATCH_SIZE).get()
-            doc_count = 0
-
+            docs = sub_collection_ref.get()
```

Comment:
I am worried that this will use up too much memory.
To strike a balance, my thought is to obtain 10 documents at a time, and continue to fetch documents into the batch if 90% of batch size is not reached.
You can reference to the function `get_number_of_unit_supplementary_data_with_dataset_id` to how to use cursor, probably it will become `sub_collection_ref.limit(10).order_by("__name__").start_after(cursor).stream()`
Make sure you test it with large dataset and trigger an exception at publish only when you also have large amount of unit data already inserted to test the functionality
Also record the time required to delete the unit data so that you will know if optimisation is needed